common/src/chain/gen_block.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Generalized block, or [GenBlock]
+
+use super::{Block, Genesis};
+use crate::primitives::Id;
+
+/// Generalized block that's either [genesis::Genesis] or [block::Block].
+///
+/// Does not contain any data, only used as `Id<GenBlock>` to signify given Id can refer to either
+/// genesis or proper block.
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub enum GenBlock {}
```

Comment:
I'm wondering whether this can be completely removed and replaced with GenBlockId...

Reply:
I don't like having extra bytes attached to the hashes, although it would simplify the code somewhat.

Reply:
I bet it'll simplify it a lot... but maybe, again, this can go to another PR (please slap a TODO on it)

Reply:
I think a sort of compromise solution could be to use `Id<GenBlock>` in data structures but `GenBlockId` could be accepted by more methods, so they have immediate access to the information whether it is a genesis id or not, without having to query chain config. We can experiment with that later I think.